requests/utils.py

```diff
@@ -932,15 +933,22 @@ def prepend_scheme_if_needed(url, new_scheme):
 
     :rtype: str
     """
-    scheme, netloc, path, params, query, fragment = urlparse(url, new_scheme)
+    parsed = parse_url(url)
+    scheme, auth, host, port, path, query, fragment = parsed
 
-    # urlparse is a finicky beast, and sometimes decides that there isn't a
-    # netloc present. Assume that it's being over-cautious, and switch netloc
-    # and path if urlparse decided there was no netloc.
+    # urlparse and parse_url determine that there isn't a netloc present in some
+    # urls. We've chosen to assume parsing is being over-cautious, and switch
+    # the netloc and path. This is maintained for backwards compatibility.
+    netloc = parsed.netloc
```

Comment:
Uhhh, what? This doesn't make any sense to me, is this something that's only needed for `urlparse`?

Reply:
Yeah, this is only a quirk for `urlparse`. I'd thought I repro'd with `parse_url` initially, but I'm unable to now.

I'll reword the whole comment for clarity. I'm not confident the test suite covers all of the initial issue though. I think I'm still in favor of leaving the swap in place as a fallback for anything we're missing.

Reply:
@sethmlarson I've reworded things to be a bit clearer.